core/src/main/scala/kafka/tools/DumpLogSegments.scala

```diff
@@ -69,7 +69,7 @@ object DumpLogSegments {
 
     CommandLineUtils.checkRequiredArgs(parser, options, filesOpt)
 
-    val print = if(options.has(printOpt)) true else false
+    val print = if(options.has(printOpt) || options.has(offsetsOpt) || options.has(valueDecoderOpt) || options.has(keyDecoderOpt)) true else false
```

Comment:
@hachikuji yeah I think `if` is still needed since there will be times you want to run this without printing log data. Unless you meant something else.  Updating the `print` val name with the next commit.